ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayFeatureLayers/DisplayFeatureLayers.cpp

```diff
@@ -0,0 +1,232 @@
+// Unpublished material - all rights reserved under the
+// Copyright Laws of the United States and applicable international
+// laws, treaties, and conventions.
+//
+// For additional information, contact:
+// Environmental Systems Research Institute, Inc.
+// Attn: Contracts and Legal Services Department
+// 380 New York Street
+// Redlands, California, 92373
+// USA
+//
+// email: contracts@esri.com
+/// \file DisplayFeatureLayers.cpp
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "DisplayFeatureLayers.h"
+
+#include "Envelope.h"
+#include "Error.h"
+#include "FeatureLayer.h"
+#include "Geodatabase.h"
+#include "GeodatabaseFeatureTable.h"
+#include "GeoPackage.h"
+#include "GeoPackageFeatureTable.h"
+#include "LayerListModel.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "MapTypes.h"
+#include "Point.h"
+#include "Portal.h"
+#include "PortalItem.h"
+#include "ServiceFeatureTable.h"
+#include "ShapefileFeatureTable.h"
+#include "SpatialReference.h"
+#include "Viewpoint.h"
+
+#include <QStandardPaths>
+
+using namespace Esri::ArcGISRuntime;
+
+// helper method to get cross platform data path
+namespace
+{
+QString defaultDataPath()
+{
+  QString dataPath;
+
+#ifdef Q_OS_IOS
+  dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+#else
+  dataPath = QStandardPaths::writableLocation(QStandardPaths::HomeLocation);
+#endif
+
+  return dataPath + "/ArcGIS/Runtime/Data/";
+}
+} // namespace
+
+DisplayFeatureLayers::DisplayFeatureLayers(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(BasemapStyle::ArcGISTopographic, this))
+{
+}
+
+DisplayFeatureLayers::~DisplayFeatureLayers() = default;
+
+void DisplayFeatureLayers::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<DisplayFeatureLayers>("Esri.Samples", 1, 0, "DisplayFeatureLayersSample");
+}
+
+MapQuickView* DisplayFeatureLayers::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void DisplayFeatureLayers::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  emit mapViewChanged();
+}
+
+void DisplayFeatureLayers::setLayerMode(FeatureLayerMode featureLayerMode)
+{
+  if (!m_map)
+    return;
+
+  m_map->operationalLayers()->clear();
```

Comment:
potential space leak - the layers are cleared from the list model but not deleted (and owned by the parent object for the lifetime of the app)